secretsms.js

```diff
@@ -0,0 +1,57 @@
+const sentenceInput = document.getElementById('sentence')
+const normalizedTextDiv = document.getElementById('normalized-text')
+const encodedMessageDiv = document.getElementById('encoded-message')
+const decodedTextDiv = document.getElementById('decoded-text')
+const encodeBtn = document.getElementById('encode-btn')
+
+encodeBtn.addEventListener('click', function () {
+  const sentence = sentenceInput.value.trim() // The .trim() method is then called on the value to remove any leading or trailing whitespace.
+
+  if (sentence.length < 50) {
+    alert('Please enter a sentence with at least 50 characters.')
+    return
+  }
+
+  const normalizedText = normalizeText(sentence) // this line calls the normalizeText fxt
+  const encodedMessage = encodeMessage(normalizedText)
+  const decodedText = decodeMessage(encodedMessage)
+
+  normalizedTextDiv.textContent = `Normalized Text: ${normalizedText}` // The ${normalizedText} is a placeholder that is replaced with the value of the normalizedText variable
+  encodedMessageDiv.textContent = `Encoded Message: ${encodedMessage}`
+  decodedTextDiv.textContent = `Decoded Text: ${decodedText}`
+})
+
+function normalizeText (sentence) {
+  return sentence.replace(/\s/g, '').toLowerCase()
+}
+
+function encodeMessage (text) {
+  const length = text.length
+  const columns = Math.ceil(Math.sqrt(length)) // calculates the number of columns required for the square pattern by taking the ceiling of the square root of the length of the text. This determines the number of characters in each column of the square.
```

Comment:
Comments on the same line like this make the code less legible

Reply:
thank you sir for the reviews, for the deployment link not working, it was a file naming error which has been handled now, and as for the comment style, it is well noted

Reply:
this issue and the other issues about the GUI, have all been fixed